### PR TITLE
Add `zwp_idle_inhibit` protocol

### PR DESCRIPTION
**Pull request by chrisduerr**

### CODE REVIEW

File: src/wayland/idle_inhibit/mod.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }
+}
+
+/// Handler trait for idle-inhibit.
+pub trait IdleInhibitHandler {
+    /// Enable idle inhibition for the output of the provided surface.
+    fn inhibit(&mut self, surface: WlSurface);
+
+    /// Stop inhibition for the provided surface.
+    fn uninhibit(&mut self, surface: WlSurface);
```
Comment (Drakulix) [inline src/wayland/idle_inhibit/mod.rs:128]: Could you add a comment describing, that this function is only called, if the client explicitly destroys the protocol object? Or maybe in simpler terms, that it is not called, if the surface is destroyed, even if it had an active inhibitor.

**Review by Drakulix — COMMENTED** [review]
LGTM, but CI doesn't seem to be happy with your doc-comment example.

File: src/wayland/idle_inhibit/inhibitor.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+//! idle-inhibit inhibitor.
+
+use _idle_inhibit::zwp_idle_inhibitor_v1::{Request, ZwpIdleInhibitorV1};
+use wayland_protocols::wp::idle_inhibit::zv1::server as _idle_inhibit;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::{Client, DataInit, Dispatch, DisplayHandle};
+
+use crate::wayland::idle_inhibit::{IdleInhibitHandler, IdleInhibitManagerState};
+
+/// State of the zwp_idle_inhibitor_v1 Global
```
Comment (PolyMeilex) [inline src/wayland/idle_inhibit/inhibitor.rs:10]: `zwp_idle_inhibitor_v1` is not a global, this is idle_inhibitor's user data.

**Review by Drakulix — APPROVED** [review]
LGTM! Thanks for protocol implementation :)
